arroyo/processing/strategies/dead_letter_queue/policies/abstract.py

```diff
@@ -1,18 +1,37 @@
 from abc import ABC, abstractmethod
-from typing import Any, Sequence
+from datetime import datetime
+from typing import Optional, Sequence, Union
+
+Serializable = Union[str, bytes]
 
 
 class InvalidMessages(Exception):
     """
     An exception to be thrown to pass bad messages to the DLQ
     so they are handled correctly.
+
+    If the original topic the message(s) were produced to is known,
+    that should be passed along via this exception.
     """
 
-    def __init__(self, messages: Sequence[Any]):
+    def __init__(
+        self,
+        messages: Sequence[Serializable],
```

Comment:
I think you will need another class here for the individual messages
That should contain at least partition and offset of the original broken message
```
# not an exception.
class InvalidMessage:
    partition: int
    offset: int
    payload: Serializable
    timestamp: DateTime
```

Reply:
It's probable the metadata of a message is not propagated throughout processing stages where this exception would be thrown.

Maybe this would be more appropriate if the offset etc is not known anymore:
```python3
@dataclass(frozen=True)
class InvalidMessage:
    payload: Serializable
    timestamp: DateTime
    reason: str = "unknown"
    original_topic: str = "unknown"
    partition: int = -1
    offset: int = -1
```

This would also continue to allow users to really pass anything to the DLQ they like. I'd still like to keep the `reason` parameter as it's probably useful for debugging messages seen in the dead letter topic.

Then it would be raised by doing something like:
```python3
invalid_message = InvalidMessage(...)
raise InvalidMessages(messages=[invalid_message])
```